### ISSUE TITLE
Allow struct names when inserting a new structure

### ISSUE THREAD

**Issue by joshwatson** [Vector35/binaryninja-api#1176]
![image](https://user-images.githubusercontent.com/5834130/47260931-55c67180-d479-11e8-9e87-945394dfa946.png)

You should be able to paste a C structure directly from a header file into this box and it work, rather than failing because the structure has a name.

**Comment by fuzyll** [comment]
As a user, I am often copying structure definitions in from random places (like, for example, moving information over from IDA or notes from a channel during a CTF). Here's some examples of things I would expect to work in various scenarios:

```
// the following, with "blah" in the "type name" box should work (this is the only thing, afaict, that actually works right now)
struct { int i; int j; };
```

```
// the following, with NOTHING in the "type name" box should work
struct blah { int i; int j; };
```

```
// the following, with "blah" in the "type name" box, should create a type called "blah" (name box overrides name in definition)
struct poop { int i; int j; };
```

```
// the following, with NOTHING in the "type name" box should work (e.g. if i copy a struct from a typedef in source - should also work if i copy the "typedef" on the front, too)
struct { int i; int j; } blah;
```

```
// the following, with "blah" in the "type name" box should work (same as above, name box overrides definition, may or may not have "typedef" on front)
struct { int i; int j; } poop;
```

**Comment by plafosse** [comment]
Fixed in 1.1.1472
